### PR TITLE
Refactor Shareableviz REST endpoint

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Split PR for https://github.com/kedro-org/kedro-viz/pull/1708

## Development notes

- Updated DeployerConfiguration class to have platform, endpoint, bucket_name as properties
- Updated /deploy REST API route to accept new DeployerConfiguration inputs
- Renamed `S3Deployer` to `AWSDeployer` to be consistent with cloud providers
- Refactored deployer classes to accept new input parameters and removed `deploy_and_get_url` method as we take endpoint (hosted url) as an input
- Updated pytests

**NOTE:** Added `# pylint: skip-file` for `/launchers/cli.py` as this PR does not include cli changes for ease of review.

## QA notes

- All pytests should pass

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

File: package/kedro_viz/integrations/deployment/aws_deployer.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+deployment class for AWS S3"""
+
+import logging
+
+import fsspec
+
+from kedro_viz import __version__
+from kedro_viz.integrations.deployment.base_deployer import BaseDeployer
+
+_S3_PROTOCOL = "s3"
```
Comment (merelcht) [inline package/kedro_viz/integrations/deployment/aws_deployer.py:11]: Does this need to be a global?

**Review by merelcht — APPROVED** [review]
LGTM! 👍 

I left two small questions, but they aren't blocking.

File: package/kedro_viz/api/rest/router.py [hunk 2]
```diff
@@ -50,11 +51,17 @@ async def get_single_pipeline_data(registered_pipeline_id: str):
...
-        deployer = S3Deployer(input_values.region, input_values.bucket_name)
-        url = deployer.deploy_and_get_url()
-        response = {"message": "Website deployed on S3", "url": url}
+        deployer = DeployerFactory.create_deployer(
+            input_values.platform, input_values.endpoint, input_values.bucket_name
+        )
+        deployer.deploy()
+        response = {
+            "message": "Website deployed on "
+            f"{input_values.platform and input_values.platform.upper()}",
```
[the remark this answers is not in the document]
Comment (ravi-kumar-pilla) [reply package/kedro_viz/api/rest/router.py:62]: platform is - aws, azure or gcp. If platform exists (not None), it resolves to uppercase (AWS etc)

File: package/kedro_viz/integrations/deployment/aws_deployer.py (hunk 1 above)
Comment (ravi-kumar-pilla) [reply package/kedro_viz/integrations/deployment/aws_deployer.py:11]: not really, but I left it as is from earlier PRs. Though we did not use the global variable in multiple functions, it is used in multiple places in the file. Also, if the protocol changes, it is easy to maintain. Do you see any issue in using a global here ?

**Review by jitu5 — APPROVED** [review]
LGTM.

**Pull request merged**